src/App.tsx

```diff
@@ -17,18 +85,37 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                handleChange={handleChange}
+                queryChange={queryChange}
+                clearQuery={clearQuery}
+                query={query}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loaderIsShown
+                ? (
+                  <Loader />
+                ) : (
+                  <TodoList
+                    todos={todos}
+                    selectedId={selectedTodo?.id}
+                    handleClick={handleClick}
+                  />
+                )}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {hasSelectedTodo && (
```

Comment:
```suggestion
      {!!selectedTodo && (
```
and remove the `hasSelectedTodo` state
General rule - no need to create a separate state value if you can get it from another state or props value